File: Dynamic Programming/Medium/LET_no926. Flip String to Monotone Increasing.py
class Solution:
    def minFlipsMonoIncr(self, s: str) -> int:
        ones = flips = 0
        for c in s:
            if c == '1':
                ones += 1
            else:
                flips += 1
            flips = min(flips, ones)
        return flips

File: Dynamic Programming/Medium/LET_no926_Flip String to Monotone Increasing.cpp
class Solution {
public:
    int minFlipsMonoIncr(string s) {
        int ones = 0, flips = 0;
        for (char c : s) {
            if (c == '1')
                ones++;
            else
                flips++;
            flips = min(ones, flips);
        }
        return flips;
    }
};